edx_proctoring/static/index.js

```diff
@@ -0,0 +1,32 @@
+export const handlerWrapper = (Handler) => {
+   let handler = new Handler({});
+
+  self.addEventListener("message", (message) => {
+    switch(message.data.type) {
+      case 'config': {
+        handler = new Handler(message.data.options);
+        break;
+      }
+      case 'startExamAttempt': {
+        if(handler.onStartExamAttempt) {
+          handler.onStartExamAttempt().then(() => self.postMessage({type: 'examAttemptStarted'}))
+        }
+        break;
+      }
+      case 'endExamAttempt': {
```

Comment:
Would also pull these message types into a constant to import here and the action handlers `actionToMessageTypesMap`

Reply:
I'm not sure I have a good mechanism for doing that. Are you proposing an es6 module with those constants, to be imported both here and in the JS that runs outside the web worker? None of the latter are es6 modules, so I'm not sure I see how that could be achieved.